File: Py-Solutions/Problem-10.py
# Find the sum of all the primes below two million.
import math

def isPrime(num):
  return sum([i for i in range(1, int(math.sqrt(num)) + 1) if num % i == 0]) == 1

def primeSum(limit):
  return sum([i for i in range(3,limit, 2) if isPrime(i)]) + 2

def main():
  print(primeSum(2000000))

if __name__ == '__main__':
  main()

File: JS-Solutions/Problem-10.js
// Find the sum of all primes below 2 million

const isPrime = (num) => {
  if (num < 2) {
    return false;
  }
  if (num == 2) {
    return true;
  }
  for(let i=2;i<=Math.sqrt(num)+1;i++){
    if(num % i == 0){
      return(false);
    }
  }
  return(true);
}

const sumOfPrimes = (limit) => {
  total = 0;
  for (let i=2;i<limit;i++) {
    if (isPrime(i)) {
      total += i;
    }
  }
  return total;
}

console.log(sumOfPrimes(2000000));
